src/Sfa.Tl.Matching.Web/Controllers/ProviderController.cs

```diff
@@ -62,52 +50,47 @@ public IActionResult Index(SearchParametersViewModel viewModel)
                 return GetIndexView(viewModel.SelectedRouteId, viewModel.Postcode);
             }
 
-            var serializedData = JsonConvert.SerializeObject(viewModel);
-            TempData[SearchParametersDataKey] = serializedData;
-            return RedirectToAction("Results");
+            return RedirectToRoute("ProviderResults_Get", new
+            {
+                routeId = viewModel.SelectedRouteId,
+                postcode = viewModel.Postcode
+            });
         }
 
         [HttpGet]
         [Route("provider-results", Name = "ProviderResults_Get")]
-        public async Task<IActionResult> Results()
+        public async Task<IActionResult> Results(int routeId, string postcode, int searchRadius = SearchParametersViewModel.DefaultSearchRadius)
```

Comment:
could you please construct special route url for this, i.e "provider-results-for-{RouteId}-within-{SearchRadius}-miles-of-{postcode}"